methods.py

```diff
@@ -72,22 +72,25 @@ def print_error(*values: object) -> None:
 
 
 def add_source_files_orig(self, sources, files, allow_gen=False):
-    # Convert string to list of absolute paths (including expanding wildcard)
     if isinstance(files, str):
+        files = [files]
+
+    for path in files:
+        _files = self.Glob(str(path))
```

Comment:
Here if `path` is already `env.{File/Entry}` or something like it is converted into string to be passed to glob and be again reconverted to `env.{File/Entry}`... It is bad but it works and have less lines...